### PR TITLE
feat: add endpoints for stop and promote

### PR DESCRIPTION
**Pull request by juanmahidalgo**

### CODE REVIEW

File: src/ports/squids/component.ts [hunk 1]
```diff
@@ -41,82 +41,100 @@ export async function createSubsquidComponent({
...
+          // there should be just one task per service
+          for (const task of tasks) {
+            squid.version = task.version || 0
+            squid.created_at = task.createdAt
+            squid.health_status = task.healthStatus
+            squid.service_status = task.lastStatus
+
+            const ip = task.attachments
+              ?.find(att => att.type === 'ElasticNetworkInterface')
+              ?.details?.find(detail => detail.name === 'privateIPv4Address')?.value
```
Comment (cyaiox) [inline src/ports/squids/component.ts:93]: should `ElasticNetworkInterface` and `privateIPv4Address` be const?

File: src/controllers/routes.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
-import { squidsHandler } from './handlers/squid-handler'
+import { listSquidsHandler, promoteSquidHandler, stopSquidHandler } from './handlers/squid-handler'
 
 // We return the entire router because it will be easier to test than a whole server
 export async function setupRouter(): Promise<Router<GlobalContext>> {
   const router = new Router<GlobalContext>()
 
-  router.get('/list', squidsHandler)
+  router.get('/list', listSquidsHandler)
+  router.get('/:id/promote', promoteSquidHandler)
```
Comment (cyaiox) [inline src/controllers/routes.ts:10]: What about using `.post` to `promote` and `stop` actions?

**Review by cyaiox — APPROVED** [review]
Stunning job! left minor comments

File: src/controllers/routes.ts (hunk 2 above)
Comment (juanmahidalgo) [reply src/controllers/routes.ts]: updated to PUT

**Review by cyaiox — APPROVED** [review]

**Pull request merged**